app/src/main/java/org/jellyfin/androidtv/ui/WatchedBadge.kt

```diff
@@ -0,0 +1,62 @@
+package org.jellyfin.androidtv.ui
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import kotlinx.android.synthetic.main.badge_watched.view.*
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.model.itemtypes.PlayableItem
+
+class WatchedBadge(context: Context) : FrameLayout(context) {
+	private var item: PlayableItem? = null
+
+	private val changeListener: () -> Unit = {
+		if (item != null) {
+			setItemData(item!!)
+		}
+	}
+
+	init {
+		val inflater = LayoutInflater.from(context)
+		inflater.inflate(R.layout.badge_watched, this)
```

Comment:
You can add a line here like `item?.let(::setItemData)` to update if you go for the primary constructor change. Could remove the secondary constructor entirely in that case.

Reply:
We still need one which takes only context as tools allegedly require them, but flipping them is a good idea.